utils.cpp

```diff
@@ -63,6 +63,13 @@ namespace marxan {
             return ss;
         }
 
+        bool is_like_numerical_data(const std::string& str)
+        {
+            size_t letter_pos = str.find_first_not_of(" ,;:^*\"/\t\'\\\n\r.+-Ee0123456789");
```

Comment:
could you write a short comment explaining this regex? Specifically the "\r.+-Ee0123456789" part and what it's trying to filter out.